src/main/kotlin/dynamic/App.kt

```diff
@@ -0,0 +1,46 @@
+package dynamic
+
+import kotlin.math.max
+
+fun main() {
+    val (N, M) = readln().split(" ").map { it.toInt() }
+
+    val activeAppCost = readln().split(" ").map { it.toInt() }
+    val inactiveAppCost = readln().split(" ").map { it.toInt() }
+
+    val appSpec = activeAppCost.mapIndexed { index, currentMemory ->
+        currentMemory to inactiveAppCost[index]
+    }
+
+    val needMemory = M - appSpec.filter { it.second == 0 }.sumOf { pair ->
+        pair.first
+    }
+    if (needMemory < 0) {
+        println(0)
+        return
+    }
```

Comment:
```suggestion
    if (needMemory <= 0) {
        println(0)
        return
    }
```
이거 = 안들어가도 되나용?! 필요한 메모리가 0일때도 종료시킬 수 있을 것 같아서요~!